hash.go

```diff
@@ -0,0 +1,115 @@
+package bn256
+
+import (
+	"crypto/sha256"
+)
+
+// Hash implements a hashing function into the G1 group. It uses Fouque-Tibouchi encoding described here
+// https://tools.ietf.org/pdf/draft-irtf-cfrg-hash-to-curve-03.pdf
+func Hash(msg []byte) *G1 {
```

Comment:
The latest version of the draft (-04) recommends using the Shallue-van de Woestijne method at Section 6.9.1.

Reply:
There first two sentences of this draft states:
"Shallue and van de Woestijne [SW06] describe a mapping that applies   to essentially any elliptic curve.  Fouque and Tibouchi [FT12] give a   concrete set of parameters for this mapping geared toward Barreto-Naehrig pairing-friendly curves [BN05]"
I updated the reference.